python/fledge/common/configuration_manager.py

```diff
@@ -751,26 +804,44 @@ async def _update_value_val(self, category_name, item_name, new_value_val):
             err_response = ex.error
             raise ValueError(err_response)
 
-    async def update_configuration_item_bulk(self, category_name, config_item_list):
+    async def update_configuration_item_bulk(self, category_name, config_item_list, request=None):
         """ Bulk update config items
 
         Args:
             category_name: category name
             config_item_list: dict containing config item values
+            request: request details to identify user info
 
         Returns:
             None
         """
-
         try:
             payload = {"updates": []}
             audit_details = {'category': category_name, 'items': {}}
             cat_info = await self.get_category_all_items(category_name)
             if cat_info is None:
                 raise NameError("No such Category found for {}".format(category_name))
+            """ Note: Update reject to the properties with permission property when the logged in user type is not
+             given in the list of permission. """
+            if request is not None:
+                if hasattr(request, "user_is_admin"):
+                    if not request.user_is_admin:
+                        from fledge.services.core.user_model import User
+                        roles = await User.Objects.get_roles()
+                        user_role_name = [r['name'] for r in roles if request.user['role_id'] == r['id']]
+                        if user_role_name:
+                            user_role_name = user_role_name[0]
+                        else:
+                            raise ValueError("No role found for logged in user.")
             for item_name, new_val in config_item_list.items():
                 if item_name not in cat_info:
                     raise KeyError('{} config item not found'.format(item_name))
+                if request is not None:
+                    if hasattr(request, "user_is_admin"):
+                        if not request.user_is_admin:
+                            if 'permission' in cat_info[item_name]:
+                                if not (user_role_name in cat_info[item_name]['permission']):
+                                    raise Exception('Forbidden')
```

Comment:
Better to add a know exception custom class, AccessDenied or PermissionDenied and handle that instead of string match

Reply:
yeah it was planned but this exception actually to be defined in user model but Not in configuration manager; due to avoid of circular dependency issue, I have kept it under general exception having message Forbidden to get the actual HTTP status code.